### DISCUSSION TITLE
Enable Credentials in server plugins.

### DISCUSSION THREAD

**Discussion by k0mar12** [nuxt/nuxt#18077] [Questions] [answered]
I have server plugin, which making server request to get some cookie

```
export default defineNuxtPlugin(async (nuxtApp) => {
    const { headers } = await $fetch.raw('http://backend.local/cookie');

    console.log(headers.get('set-cookie'));
});
```

Response has needed cookie, but how i can pass them to the client?
setCookie works only for server api routes.

In nuxt 2 i just had server plugin, which has been injected to axios interceptors:

```
export default function ({ $axios, res }) {
    $axios.onResponse(response => {
        res.setHeader('Set-Cookie', response.headers['set-cookie']);
    });
}
```
That was enough, **axios** do it by himself, what about **ohmyfetch** ?

**Comment by aligzl** [comment]
+1

**Comment by Diizzayy** [comment, marked as answer]
@Komar12124 
> setCookie works only for server api routes

Actually, setCookie() can be called by server-side plugins / composables, you just have to pass the res from ssrContext

> Response has needed cookie, but how i can pass them to the client?

I've rewritten your code below (tested & works as you'd expect) 👍🏽
```
import { setCookie } from "h3";

export default defineNuxtPlugin(async ({ ssrContext }) => {
  const { headers } = await $fetch.raw("http://backend.local/cookie");
  
  const cookieHeader = headers.get("set-cookie");

  const regex = /(?<cookieName>[^=]+)=(?<cookieValue>[^;]+); (?<options>.*)/;
  const { cookieName, cookieValue } = cookieHeader.match(regex).groups;

  setCookie(ssrContext.res, cookieName, cookieValue);
});
```

**Reply by Diizzayy** [reply]
@Komar12124  mark this as closed if this response helped

**Reply by k0mar12** [reply]
@Diizzayy Great idea to ssrContext, thank You.

I made some changes, with external lib **set-cookie-parser** for fully correct parsing of cookie with all options.
Also, i covered case, when response has many records of cookie

```
import { defineNuxtPlugin } from '#app';
import { setCookie } from 'h3';
import { splitCookiesString, parse } from 'set-cookie-parser';

/**
 * Provide cookie for client side from server request
 *
 */
export default defineNuxtPlugin(async ({ ssrContext }) => {
    const { headers } = await $fetch.raw('http://backend.local');

    for (const cookie of parse(splitCookiesString(headers.get('set-cookie')))) {
        setCookie(ssrContext.res, cookie.name, cookie.value, cookie);
    }
});

```

**Reply by Diizzayy** [reply]
This is a good solution.
